test/ClientShutdownTest.js

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+
+var chai = require('chai');
+var expect = chai.expect;
+var chaiAsPromised = require('chai-as-promised');
+chai.use(chaiAsPromised);
+
+var sinon = require('sinon');
+var sandbox = sinon.createSandbox();
+
+var RC = require('./RC');
+var HzErrors = require('../.').HazelcastErrors;
+var {Client} = require('../.');
+var {RepairingTask} = require('../lib/nearcache/RepairingTask');
+var {NearCacheManager} = require('../lib/nearcache/NearCacheManager');
+var {Statistics} = require('../lib/statistics/Statistics');
+var {ClientConnectionManager} = require('../lib/invocation/ClientConnectionManager');
+var {InvocationService} = require('../lib/invocation/InvocationService');
+var {PartitionService} = require('../lib/PartitionService');
+var {Heartbeat} = require('../lib/HeartbeatService');
+var {ListenerService} = require('../lib/ListenerService');
+
+describe('ClientShutdownTest', function () {
+
+    var cluster;
+
+    afterEach(async function () {
+        sandbox.restore();
+        if (cluster != null) {
+            await RC.terminateCluster(cluster.id);
+            cluster = null;
+        }
+    });
+
+    it('client should call shutdown on failed start', async function () {
```

Comment:
Node.js client v3.12.x is supposed to support Node.js 4+, so we can't use async/await syntax in tests. Could you update the test in Promise callbacks syntax?